### PR TITLE
Blazor JS Interop example enhancements

### LINKED ISSUE #15140: JS function for "setElementValue" (on <Detect when a Blazor app is prerendering> section)
Hey,
I know this is not so tricky ;) but just for the sake of the documentation :), no JavaScript function has provided for "setElementValue" so it leads to "Nothing" on the &lt;Detect when a Blazor app is prerendering&gt; section, and raising an error for the next part.

wish the best.
 

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 49807900-0d56-e23c-1aef-d79099caeffb
* Version Independent ID: feff434f-912c-0985-d575-4a23a629057d
* Content: [ASP.NET Core Blazor JavaScript interop](https://docs.microsoft.com/en-us/aspnet/core/blazor/javascript-interop?view=aspnetcore-3.0#detect-when-a-blazor-app-is-prerendering)
* Content Source: [aspnetcore/blazor/javascript-interop.md](https://github.com/aspnet/AspNetCore.Docs/blob/master/aspnetcore/blazor/javascript-interop.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-blazor**
* GitHub Login: @guardrex
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #15140

**Ok ... ready now for review. I made another round of small updates this morning (Thur, 10/17).**

[Internal Review Topic](https://review.docs.microsoft.com/en-us/aspnet/core/blazor/javascript-interop?view=aspnetcore-3.0&branch=pr-en-us-15143)

The ask is to improve the examples for JS Interop:

* Show the JS functions involved even if they aren't germane to the focus of the section's content because:
  * Devs **_DO_** like to cut-'n-paste code bits into sample apps to play with the bits. Providing these is low cost with plenty of upside for dev efficiency.
  * It also allows us to reinforce an `InvokeAsync`/`InvokeVoidAsync` concept: JS functions return/don't return values, respectively.
* I'm also going work on consistency in JS function naming: These updates go with the camel case everywhere.

Not quite 100% there tho on this one (*javascript-interop\samples_snapshot\inject-abstraction.razor*) ...

```cshtml
@inject IJSRuntime JSRuntime

@code {
    protected override void OnInitialized()
    {
        StocksService.OnStockTickerUpdated += stockUpdate =>
        {
            JSRuntime.InvokeVoidAsync("handleTickerChanged",
                stockUpdate.symbol, stockUpdate.price);
        };
    }
}
```

What's the story there on **_not_** awaiting the call to `InvokeVoidAsync`? Should it await, then `async stockUpdate`? ... or is this ok in its present form?

Thanks @aminm1364 🏍 and @Legends 🎸

### CODE REVIEW

File: aspnetcore/includes/blazor-prerendering.md [hunk 1]
```diff
@@ -11,17 +11,25 @@ To delay JavaScript interop calls until after the connection with the browser is
...
         }
     }
 }
 ```
 
+For the preceding example code, provide a `setElementValue` JavaScript function inside the `<head>` element of *wwwroot/index.html* (Blazor WebAssembly) or *Pages/_Host.cshtml* (Blazor Server). The function is called with `IJSRuntime.InvokeVoidAsync` and doesn't return a value:
+
+```html
+<script>
+  window.setElementValue = (elem, val) => elem.value = val;
```
Comment (danroth27) [inline aspnetcore/includes/blazor-prerendering.md]: I'm worried that this is showing an anti-pattern. By manipulating the DOM directly in a way that Blazor is unaware of, you run the risk of messing up Blazor's diffing algorithm. I better example might be initializing some sort of JS component, but I don't know what a simple example might look like. @pranavkm @javiercn @SteveSandersonMS any ideas?

Comment (guardrex) [reply aspnetcore/includes/blazor-prerendering.md]: We can always add a note that this is for demo purposes only. We've done that a few times with good outcomes AFAIK. I recommend an explicit remark in the topic and a code comment to that effect.

**Comment by guardrex** [comment]
Plz don't forget about my ask in the opening comment on async in that scenario. :point_up:

File: aspnetcore/includes/blazor-prerendering.md (hunk 1 above)
Comment (danroth27) [reply aspnetcore/includes/blazor-prerendering.md]: Yeah, I would suggest two things: 1. change the DOM element from an input to a static div (so folks don't try to bind to it or do other Blazor-y things), and 2. Add a note warning folks that modifying the DOM directly with JS can interfere with Blazor's change tracking.

**Comment by danroth27** [comment]
> Plz don't forget about my ask in the opening comment on async in that scenario. ☝️

I reviewed that example with @pranavkm and we think the call to `InvokeVoidAsync` in the event handler is fine as is.

**Comment by guardrex** [comment]
@danroth27 Examples updated to use `<div>` seems nice ... and the WARNING notes are added in both spots.

**Comment by guardrex** [comment]
@danroth27 Wait for Javier to look or go ahead and merge this?

**Pull request merged**